File: number theory/totient_extreme.cpp
#include <stdio.h>
#define MAX 10001

long long fi[MAX], ps[MAX], res;
int tests, n;

void FillEuler(void){
    int i, j;
    fi[0] = fi[1] = 0;
    for(i = 2; i < MAX; i++)
        if(!fi[i])
            for(j = i; j < MAX; j += i){
                if(!fi[j]) fi[j] = j;
                fi[j] -= fi[j]/i;
            }

    ps[0] = 0; ps[1] = 1;
    for(i = 2; i < MAX; i++)
        ps[i] = ps[i-1] + fi[i];
}

int main(void){

    FillEuler();

    scanf("%d",&tests);

    while(tests--){
        scanf("%d",&n);
        res = ps[n] * ps[n];
        printf("%lld\n",res);
    }
    return 0;
}

File: chatgpt code/Totient Extreme.py
MAX = 10001

fi = [0] * MAX
ps = [0] * MAX

def fill_euler():
    global fi, ps
    for i in range(2, MAX):
        if fi[i] == 0:
            for j in range(i, MAX, i):
                if fi[j] == 0:
                    fi[j] = j
                fi[j] -= fi[j] // i
    
    ps[0] = 0
    ps[1] = 1
    for i in range(2, MAX):
        ps[i] = ps[i - 1] + fi[i]


fill_euler()
tests = int(input())
for _ in range(tests):
    n = int(input())
    res = ps[n] * ps[n]
    print(res)
